src/main/scala/io/scalac/tezos/translator/model/types/UUIDs.scala

```diff
@@ -0,0 +1,47 @@
+package io.scalac.tezos.translator.model.types
+
+import java.util.UUID
+import io.estatico.newtype.macros.newtype
+import io.circe.{Decoder, DecodingFailure, Encoder, HCursor}
+import slick.ast.BaseTypedType
+import slick.jdbc.JdbcType
+import slick.jdbc.PostgresProfile.api._
+import sttp.tapir.{Schema, SchemaType}
+import io.circe.syntax._
+import cats.syntax.either._
+import cats.syntax.option._
+import scala.reflect.ClassTag
+import Util._
+
+object UUIDs {
+
+  @newtype case class LibraryEntryId(v: UUID)
```

Comment:
Previous implementation (`Uid`) was a sealed abstract case class. This made quite strong guarantees that it could be constructed only through given methods, therefore we were sure that every `Uid` instance had unique UUID.

With `LibraryEntryId` it is possible to construct it in two ways - with case class apply and generateLibraryEntryId.
In particular it allowed to avoid such a situation when an exception will not be caught:
```LibraryEntryId(UUID.fromString("blabla"))```

I admit I liked this bound on constructors.  Do you think that this restriction of constructor is not needed ? WDYT ?

Reply:
We still have the same guarantees about the uniqueness of UUID. About constructor, I think we still don't have real guarantees with `Uid` too, we can just do something like - `Uid.fromString("blablabla").toOption.get`, and get a more weird error message here, I mean with `Uid` we just wrap that exception to `Try` and that all. I think it's can work if we will add some fallback mechanism like 
```
def fromString(s: String): Try[Uid] =
            new Uid(
              UUID.fromString(s)
              .toOption
              .getOrElse {
              log.error("oh no, can't parse UUID")
              UUID.randomUUID }
) {})
```
then we will have some guarantees but we will lose in logic, also I think we need control of parsing that UUID at the edge of the world, we can use UUIDString from the refined library to guarantee that string is proper UUID and use it instead of java UUID, and if you don't mind I'll add it tomorrow

Reply:
>  UUIDString from the refined library
Sounds lovely ;)